galaxy/wrapper/bamCompare.xml

```diff
@@ -195,11 +194,91 @@
+        <test expect_num_outputs="1">
+            <param name="bamFile1" value="testA.bam" ftype="bam"/>
+            <param name="bamFile2" value="testA.cram" ftype="cram"/>
+            <param name="showAdvancedOpt" value="yes"/>
+            <param name="outFileFormat" value="bedgraph"/>
+            <param name="binSize" value="10"/>
+            <param name="type" value="ratio"/>
+            <output name="outFileName" ftype="bedgraph">
+                <assert_contents>
+                    <has_text_matching expression="3R\t0\t200\t1"/>
+                    <has_text_matching expression="chr_cigar\t0\t200\t1"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!-- Test with BAM and CRAM files and substract option-->
+        <test expect_num_outputs="1">
+            <param name="bamFile1" value="testA.bam" ftype="bam"/>
+            <param name="bamFile2" value="testA.cram" ftype="cram"/>
+            <param name="showAdvancedOpt" value="yes"/>
+            <param name="outFileFormat" value="bedgraph"/>
+            <param name="binSize" value="10"/>
+            <param name="type" value="substract"/>
+            <output name="outFileName" ftype="bedgraph">
+                <assert_contents>
+                    <has_text_matching expression="3R\t0\t200\t0"/>
+                    <has_text_matching expression="chr_cigar\t0\t200\t0"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!-- Test with BAM files with pseudocounts-->
+        <test expect_num_outputs="1">
+            <param name="bamFile1" value="testA.bam" ftype="bam"/>
+            <param name="bamFile2" value="testB.bam" ftype="bam"/>
+            <param name="showAdvancedOpt" value="yes"/>
+            <param name="outFileFormat" value="bedgraph"/>
+            <param name="binSize" value="10"/>
+            <param name="type" value="ratio"/>
+            <param name="pseudocount" value="1 1"/>
+            <output name="outFileName" ftype="bedgraph">
+                <assert_contents>
+                    <has_text_matching expression="3R\t0\t50\t1"/>
+                    <has_text_matching expression="3R\t50\t100\t0.571429"/>
+                    <has_text_matching expression="3R\t100\t150\t1.14286"/>
+                    <has_text_matching expression="3R\t150\t200\t0.8"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!-- Test with BAM files with skipZeroOverZero -->
+        <test expect_num_outputs="1">
+            <param name="bamFile1" value="testA.bam" ftype="bam"/>
+            <param name="bamFile2" value="testB.bam" ftype="bam"/>
+            <param name="showAdvancedOpt" value="yes"/>
+            <param name="outFileFormat" value="bedgraph"/>
+            <param name="binSize" value="10"/>
+            <param name="type" value="ratio"/>
+            <param name="skipZeroOverZero" value="--skipZeroOverZero"/>
```

Comment:
boolean params (I think this is one) should be tested with `value="true"`/`value="false"` instead of the eventual values.
This is a more stringent test because it checks also whether you typed the correct true/falsevalue in the input section.